goldstone/core/models.py

```diff
@@ -2634,3 +2636,70 @@ def integration(cls):
     def resourcetype(cls):
 
         return "routers"
+
+
+class DefinedSearch(models.Model):
+    """Defined searches, both system and user-created."""
+
+    # This object's Goldstone UUID.
+    uuid = UUIDField(version=1, auto=True, primary_key=True)
+
+    name = models.CharField(max_length=64)
+
+    description = models.CharField(max_length=1024, default='Defined Search')
+
+    query = models.CharField(max_length=512,
+                             help_text='YAML Elasticsearch query body')
+
+    protected = models.BooleanField(default=False,
+                                    help_text='True if this is system-defined')
+
+    created = CreationDateTimeField(editable=False, blank=True)
+    updated = ModificationDateTimeField(editable=True, blank=True)
+
+    @classmethod
+    def field_has_raw(cls, _):
+        """Bypass the "field has raw representation" check in
+        drfes.models.DailyIndexDocType."""
+
+        return False
+
+    @classmethod
+    def execute_query(cls, name):
+        """ executes an elasticsearch query and returns the result of that query.
+        :param query:
+        :return:
+        """
```

Comment:
Call me Mr. Docstring Nit-picker...

The docstring should be an English command.  Per https://www.python.org/dev/peps/pep-0257/:

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

This docstring should be:

    """Execute an Elasticsearch query and return the result.

    :param query:  fill this in.
    :type query: fill this in.
    :return: fill this in.
    :rtype: fill this in.

    """

If the :return: or :rtype: is obvious, you can omit them.

Same comment goes for EventQueryDef, below.